src/Input/Select/Select.tsx

```diff
@@ -15,346 +15,271 @@ import {
-        });
-      }
-    } else if (isLoading && isLoading !== prevProps.isLoading) {
-      this.setState({
-        isLoading: isLoading,
-      });
-    }
-  }
+    document.addEventListener('click', handleOnBlur, false);
 
-  componentWillUnmount() {
-    document.removeEventListener('click', this.handleOnBlur, false);
-  }
+    return () => document.removeEventListener('click', handleOnBlur, false);
+  }, []);
 
-  handleOnBlur = (event: MouseEvent) => {
-    const element = event.target as HTMLElement;
-    if (!ReactDOM.findDOMNode(this.node.current).contains(element)) {
-      this.setState({
-        isFocus: false,
-      });
+  React.useEffect(() => {
+    if (value && value !== defaultValue) {
+      setSelectedValue(value as string);
+      setDefaultValue(value as string);
+      setFloating(true);
+    } else if (value === '') {
+      setSelectedValue(value);
+      setDefaultValue(value);
+      setFloating(false);
     }
-  };
+  }, [value]);
 
-  handleFocusOut = (
-    onBlur: (e: React.FocusEvent<HTMLInputElement>) => void
-  ) => {
-    const listener = (e: React.FocusEvent<HTMLInputElement>) => {
-      this.setState({
-        floating: e.target.value.length > 0,
-      });
+  React.useEffect(() => {
+    setFilterValue(React.Children.map(children, data => data));
+  }, [children, value]);
+
+  const handleFocusOut = React.useCallback(
+    (e: React.FocusEvent<HTMLInputElement>) => {
+      setFloating(e.target.value.length > 0);
 
       if (onBlur !== undefined) {
         return onBlur(e);
       }
-    };
-
-    return listener;
-  };
+    },
+    []
+  );
 
-  handleFocus = (onFocus: (e: React.FocusEvent<HTMLInputElement>) => void) => {
-    const listener = (e: React.FocusEvent<HTMLInputElement>) => {
-      this.setState({
-        isFocus: true,
-      });
+  const handleFocus = React.useCallback(
+    (e: React.FocusEvent<HTMLInputElement>) => {
+      setIsFocus(true);
 
       if (onFocus !== undefined) {
         return onFocus(e);
       }
-    };
-
-    return listener;
-  };
+    },
+    []
+  );
 
-  handleChange = (
-    onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
-  ) => {
-    const listener = (e: React.ChangeEvent<HTMLInputElement>) => {
-      const { children } = this.props;
+  // Should be called when the user types into the input
+  const handleChange = React.useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
       const childrenArray = React.Children.toArray(children) as Array<
         React.ReactElement<SelectItemProps>
       >;
 
-      this.setState({
-        selectedValue: e.target.value,
-        filterValue: childrenArray.filter(data =>
+      setSelectedValue(e.target.value);
+      setFilterValue(
+        childrenArray.filter(data =>
           data.props.children
             .toLowerCase()
             .includes(e.target.value.toLowerCase())
-        ),
-        cursor: 0,
-      });
-
-      if (onChange !== undefined) {
-        return onChange(e);
```

Comment:
same as https://github.com/glints-dev/glints-aries/pull/252#discussion_r393438409

Reply:
The funny thing is, implementations seem to depend on the changed value being returned here. For example here: 
![image](https://user-images.githubusercontent.com/3341028/77391478-ce023680-6dd3-11ea-8e3c-dca0a9c78170.png) (app/modules/UgcQnA/Components/QuestionForm/TopicField.js in DST)

Reply:
Yes, it will be handy if onChange and onOptionClick can return the changed value only.

Do you think we should refactor the use of Select on dst and employers for ^?

Reply:
I tested agains some sample DST usages. I think the current state of the PR preserves the previous behaviour. As evidenced by the screenshot above, the `onChange` component did return the changed value.